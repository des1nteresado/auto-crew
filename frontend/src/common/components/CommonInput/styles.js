const styles = (theme) => ({
  root: {
    width: '100%',
    '& .MuiInput-root': {
      height: 40,
    },
    '& .Mui-error': {
      color: `${theme.palette.error.main} !important`,
      '& .MuiInputBase-input': {
        color: theme.palette.error.main,
        borderColor: theme.palette.error.main,
      },
    },
    '& .MuiInput-underline:after': {
      display: 'none',
    },
    '& .MuiInput-underline:before': {
      display: 'none',
    },
    '& .MuiFormLabel-root.Mui-focused': {
      color: theme.palette.text.primary,
    },
    '& label + .MuiInput-formControl': {
      marginTop: 0,
    },
    '& .MuiInputBase-input': {
      fontSize: 13,
      border: '1px solid',
      borderColor: theme.palette.text.primary,
      borderRadius: 5,
      paddingTop: 12,
      paddingLeft: 14,
      color: theme.palette.black,
    },
    '& .MuiInputBase-input:focus': {
      borderColor: theme.palette.black,
    },
    '& .MuiInputLabel-formControl': {
      fontSize: 13,
      left: 15,
      top: 4,
      color: theme.palette.text.primary,
      transform: 'translate(0, 9px) scale(1)',
    },
    '& .MuiInputLabel-shrink': {
      color: theme.palette.text.primary,
      transform: 'translate(0, 2px) scale(0.75)',
    },
    '& .MuiSelect-select:focus': {
      backgroundColor: 'transparent',
    },
    '& .MuiSelect-icon': {
      color: theme.palette.text.primary,
      right: 10,
    },
    '& .MuiFormHelperText-root': {
      position: 'absolute',
      bottom: -18,
    },
  },
});

export default styles;
