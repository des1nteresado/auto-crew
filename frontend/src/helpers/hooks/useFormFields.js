import { useState } from 'react';

export const useFormFields = (initialState) => {
  const [fields, setValue] = useState(initialState);

  return [
    fields,
    (event) => {
      if (event.target.type === 'checkbox') {
        setValue({
          ...fields,
          [event.target.name]: event.target.checked,
        });
      } else {
        setValue({
          ...fields,
          [event.target.name]: event.target.value,
        });
      }
    },
  ];
};
