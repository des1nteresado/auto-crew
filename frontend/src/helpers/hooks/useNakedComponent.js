export const useNakedComponent = (component) => {
  return component?.type?.Naked;
};
