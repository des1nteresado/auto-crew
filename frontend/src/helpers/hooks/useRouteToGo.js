import { useCallback } from 'react';
import { useHistory } from 'react-router-dom';

export const useRouteToGo = (path) => {
  const history = useHistory();

  return useCallback(() => history.push(path), [path, history]);
};
