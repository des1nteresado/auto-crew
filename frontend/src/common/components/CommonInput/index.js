import React, { useState, useCallback } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { v1 as uuid } from 'uuid';
import { TextField, withStyles } from '@material-ui/core';

import styles from './styles.js';

const CommonInput = ({
  value,
  name,
  style,
  error,
  classes,
  multiline,
  helperText,
  label,
  isTouchedByDefault,
  ...otherProps
}) => {
  const inputId = `${name}-${uuid()}`;

  const [isTouched, setIsTouched] = useState(false);

  const onBlur = useCallback(() => {
    setIsTouched(true);
  }, [setIsTouched]);

  return (
    <TextField
      value={value}
      classes={{
        root: clsx(classes.root, style && style),
      }}
      name={name}
      id={inputId}
      label={label}
      autoComplete="off"
      onBlur={onBlur}
      multiline={multiline}
      error={(isTouched || isTouchedByDefault) && error}
      helperText={
        !multiline ? (isTouched || isTouchedByDefault) && error && helperText : helperText
      }
      {...otherProps}
    />
  );
};

CommonInput.defaultProps = {
  name: uuid(),
  value: '',
  label: '',
  style: '',
  helperText: '',
  type: 'text',
  error: false,
  multiline: false,
  isTouchedByDefault: false,
  onChange: () => {},
};

CommonInput.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  onChange: PropTypes.func,
  value: PropTypes.string,
  name: PropTypes.string,
  helperText: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  label: PropTypes.string,
  error: PropTypes.bool,
  multiline: PropTypes.bool,
  isTouchedByDefault: PropTypes.bool,
  style: PropTypes.string,
  type: PropTypes.string,
};

export default React.memo(withStyles(styles)(CommonInput));
