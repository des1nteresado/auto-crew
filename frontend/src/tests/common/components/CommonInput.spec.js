import React from 'react';
import { mount, shallow } from 'enzyme';

import CommonInput from 'common/components/CommonInput';
import { useNakedComponent } from 'helpers/hooks/useNakedComponent';

const defaultProps = {
  isTouchedByDefault: false,
  error: true,
  multiline: false,
  helperText: 'error text',
  style: 'styleClass',
};

jest.mock('uuid', () => ({
  v1: () => 'test-id',
}));

const setUp = (props) => shallow(<CommonInput {...props} {...defaultProps} />);

describe('CommonInput', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders CommonInput component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('shows error helperText when error prop is passed and input is touched', () => {
    const commonInput = mount(<CommonInput {...defaultProps} />);
    const input = commonInput.find('input');

    input.simulate('blur');
    const p = commonInput.find('p');
    expect(p.text()).toBe(defaultProps.helperText);
  });

  it('shows helperText when multiline prop is true', () => {
    const commonInput = mount(<CommonInput {...defaultProps} multiline />);
    const p = commonInput.find('p');

    expect(p.text()).toBe(defaultProps.helperText);
  });

  describe('default props', () => {
    const nakedComponent = useNakedComponent(CommonInput);

    it('should use default prop onChange', () => {
      const result = nakedComponent.defaultProps.onChange();
      expect(result).toBe(undefined);
    });
  });
});
