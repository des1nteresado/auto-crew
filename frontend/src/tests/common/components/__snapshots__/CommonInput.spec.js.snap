// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonInput renders CommonInput component correctly 1`] = `
<CommonInput
  classes={
    Object {
      "root": "CommonInput-root-1",
    }
  }
  error={true}
  helperText="error text"
  isTouchedByDefault={false}
  label=""
  multiline={false}
  name="test-id"
  onChange={[Function]}
  style="styleClass"
  type="text"
  value=""
/>
`;
